src/Generator/Passes/DelegatesPass.cs

```diff
@@ -187,6 +187,15 @@ private DeclarationContext GetDeclContextForDelegates(DeclarationContext @namesp
                     ).GroupBy(d => d.Name).Where(g => g.Any(d => d.HasDeclarations)).ToList();
                 if (groups.Count == 1)
                     parent = groups.Last().Last();
+                else
+                {
+                    foreach (var g in groups)
+                    {
+                        parent = g.ToList().Find(ns => ns.Name == module.LibraryName || ns.Name == module.OutputNamespace);
```

Comment:
For a test, you could add a method which takes a function pointer, or a virtual method, to a test binding where we do not define an output name-space, just like in your case. The function pointer or a virtual method would force the generation of a delegate so the issue should be reproduced.
In general, I would really like to just merge your code and make it "easy" to both you and me. Unfortunately, our system covers an extremely large area (C++) and the lack of tests is guaranteed to allow regressions in the long term. To this very day creating a test case is by far my most unpleasant part of the work but it is simply necessary. I really hope you understand.

Reply:
Completely agreed. Good test suite is essential for project like CppSharp. I would like to actually step it up and suggest adding more CI test builds. There could easily be tests for GCC 4.8 / 5 / 6 / and maybe 7 added, including LLVM builds by CI. To compensate long build times LLVM builds can be cached as well.

Reply:
One more thing - i have patch in the works that changes llvm package name, adding extra info for linux builds. Possible names with the patch:

* `llvm-9bde5f-linux-x64-gcc-4.8-Release`
* `llvm-9bde5f-linux-x64-gcc-4.9-Debug`
* `llvm-9bde5f-linux-x64-gcc-5-Debug`
* `llvm-9bde5f-linux-x64-gcc-6-Release-no-cxx11`

It would make it easier to support multiple compiler versions on linux as well as pave the way for testing multiple compiler versions on travis.

Reply:
The idea is more than brilliant but the problem is pure manpower. We need to update Clang every now and then (we try doing it twice a year) and it's cumbersome enough as it is. 3 packages to build each time, Windows, macOS and Linux, and your suggestion would make them 7. In addition we'd also have to maintain different parser bindings for each. I simply cannot imagine who could take responsibility for all of this grunt work.

Reply:
Considering small coverage of linux binary builds you could even retire them altogether. Linux users can build LLVM locally. As for bindings - there is no need to maintain more bindings than supported at the moment. You have linux bindings generated with gcc 4.8 right? I generated them with gcc 7.2. The [difference](https://github.com/rokups/CppSharp/commit/b9a0abd41cf5b4af6d8812a6716fa5d827a01436) is minimal and that code is probably not even used by CppSharp, so you can keep maintaining bindings for the single compiler version that is officially supported. So all extra CI jobs would do is give wider test coverage. All of these compilers are supposed to work now anyway.